tests/remote_test.sh

```diff
@@ -141,59 +141,18 @@ function test_cmd_remote()
   assertEquals "cmd_remotely should not work ($ID)" "$expected_command" "$output"
 }
 
-function test_cp_host2remote()
+function test_cp2remote()
 {
   local src="/any/path"
   local dst="/any/path/2"
-  local remote="172.16.224.1"
-  local port="2222"
-  local user="kw"
   local flag="TEST_MODE"
   local ID
 
-  ID=1
-  expected_command="rsync -e 'ssh -p $port' -La $src $user@$remote:$dst --rsync-path='sudo rsync'
-ssh -p $port ${user}@${remote} sudo \"chown -R root:root $dst\""
-  output=$(cp_host2remote "$src" "$dst" "$remote" "$port" "$user" "$flag")
-  assertEquals "Command did not match ($ID)" "$expected_command" "$output"
-
-  ID=2
-  src="$KW_CACHE_DIR/$LOCAL_TO_DEPLOY_DIR/*"
-  expected_command="rsync -e 'ssh -p $port' -La $src $user@$remote:$dst --rsync-path='sudo rsync'
-ssh -p $port ${user}@${remote} sudo \"chown -R root:root $dst\""
-  output=$(cp_host2remote "" "$dst" "$remote" "$port" "$user" "$flag")
-  assertEquals "Command did not match ($ID)" "$expected_command" "$output"
-
-  ID=3
-  src="$KW_CACHE_DIR/$LOCAL_TO_DEPLOY_DIR/*"
-  dst="$REMOTE_KW_DEPLOY"
-  expected_command="rsync -e 'ssh -p $port' -La $src $user@$remote:$dst --rsync-path='sudo rsync'
-ssh -p $port ${user}@${remote} sudo \"chown -R root:root $dst\""
-  output=$(cp_host2remote "" "" "$remote" "$port" "$user" "$flag")
-  assertEquals "Command did not match ($ID)" "$expected_command" "$output"
+  populate_remote_info
 
-  ID=4
-  src="$KW_CACHE_DIR/$LOCAL_TO_DEPLOY_DIR/*"
-  dst="$REMOTE_KW_DEPLOY"
-  expected_command="rsync -e 'ssh -p $port' -La $src $user@localhost:$dst --rsync-path='sudo rsync'
-ssh -p $port ${user}@localhost sudo \"chown -R root:root $dst\""
-  output=$(cp_host2remote "" "" "" "$port" "$user" "$flag")
-  assertEquals "Command did not match ($ID)" "$expected_command" "$output"
-
-  ID=5
-  src="$KW_CACHE_DIR/$LOCAL_TO_DEPLOY_DIR/*"
-  dst="$REMOTE_KW_DEPLOY"
-  expected_command="rsync -e 'ssh -p 22' -La $src $user@localhost:$dst --rsync-path='sudo rsync'
-ssh -p 22 ${user}@localhost sudo \"chown -R root:root $dst\""
-  output=$(cp_host2remote "" "" "" "" "$user" "$flag")
-  assertEquals "Command did not match ($ID)" "$expected_command" "$output"
-
-  ID=6
-  src="$KW_CACHE_DIR/$LOCAL_TO_DEPLOY_DIR/*"
-  dst="$REMOTE_KW_DEPLOY"
-  expected_command="rsync -e 'ssh -p 22' -La $src root@localhost:$dst --rsync-path='sudo rsync'
-ssh -p 22 root@localhost sudo \"chown -R root:root $dst\""
-  output=$(cp_host2remote "" "" "" "" "" "$flag")
```

Comment:
These tests are stressing this function by removing the optional parameters and check their behave. I still think this has value because you still using the original parameters.